### ISSUE TITLE
can't find package in monorepo provided by submodule

### ISSUE THREAD

**Issue by trim21** [renovatebot/renovate#18203]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, tell us what version of Renovate you run.

32.222.3

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

I have a monorepo, one of it's package is a git submodule, and I have a yarn workspaces config in my package.json.

and I have a config `"cloneSubmodules": true`, and submodule is cloned in log.

All packages are detected correctly instead of this git submodule.

reproduction: https://github.com/trim21/cactbot-triggers-src/issues/2

### Relevant debug logs

https://app.renovatebot.com/dashboard#github/trim21/cactbot-triggers-src/842159188

### Have you created a minimal reproduction repository?

No reproduction, but I have linked to a public repo where it occurs

**Comment by rarkins** [comment]
It's not clear to me what you're expecting to happen.

**Comment by trim21** [comment]
> It's not clear to me what you're expecting to happen.

In [dashboarad](https://github.com/trim21/cactbot-triggers-src/issues/2) you can find a warning message 

> Renovate failed to look up the following dependencies: `cactbot`.

And the monorepo package `cactbot` is in a git submodule, renovatebot always try to lookup it in npm.

Another monorepo packge are not in git submodule are detected fine.

**Comment by rarkins** [comment]
It's line 32 of triggers/package.json and looks just like any other dependency 

What logic do you think Renovate should use to know that this shouldn't be looked up on npm?

**Comment by trim21** [comment]
You are right, but line 31, the `overlay-plugin` is also a monorepo package and it works. I'm hoping the submodules monorepo package works like non submodules package.



https://github.com/trim21/cactbot-triggers-src/blob/master/triggers/package.json#L31

https://github.com/trim21/cactbot-triggers-src/blob/master/overlay_plugin/package.json#L2

**Comment by trim21** [comment]
{
            "currentValue": "^0.0.0",
            "datasource": "npm",
            "depIndex": 0,
            "depName": "overlay-plugin",
            "depType": "dependencies",
            "isInternal": true,
            "prettyDepType": "dependency",
            "skipReason": "internal-package",
            "updates": []
          },

{
            "currentValue": "^0.27.12",
            "datasource": "npm",
            "depIndex": 1,
            "depName": "cactbot",
            "depType": "dependencies",
            "prettyDepType": "dependency",
            "versioning": "npm",
            "warnings": [
              {
                "topic": "cactbot",
                "message": "Failed to look up dependency cactbot"
              }
            ],
            "updates": []
          },
